tests/unit/test_graph.cpp

```diff
@@ -986,7 +981,10 @@ TEST_F(GraphTest, GraphStatistics) {
 	ASSERT_EQ(edge_deleted, 3);
 
 	// delete disconnected node
-	Graph_DeleteNode(g, &n[3]);
+	Graph_AcquireWriteLock(g);
```

Comment:
lock is acquired at line 971, do we really need to acquire and release multiple times in this function ?